src/test/qgstest.h

```diff
@@ -292,13 +296,74 @@ class TEST_EXPORT QgsTest : public QObject
       return result;
     }
 
+    /**
+     * For internal use only -- use QGSCOMPARELONGSTR macro instead.
+     */
+    bool checkLongStr( const char *file, const char *function, int line, const QString &name, const QString &referenceName, const QByteArray &actualStr )
+    {
+      QString subPath = "control_files/" + mControlPathPrefix + "/expected_" + name + "/" + "expected_" + referenceName;
+      QString expectedPath = testDataPath( subPath );
+      QFile expectedFile( expectedPath );
+      if ( ! expectedFile.open( QFile::ReadOnly  | QIODevice::Text ) )
+      {
+        qWarning() << "Unable to open expected data file" << expectedPath;
+        return false;
+      }
+      QByteArray expectedStr = expectedFile.readAll();
+
+      if ( actualStr.size() != expectedStr.size() )
+      {
+        qWarning() << "Array have not the same length (actual vs expected):" << actualStr.size() << "vs" << expectedStr.size();
```

Comment:
Shouldn't this fail? Eg if the actual string is a truncated version of expected, then the test will currently pass...

Reply:
from what I have tested, this works fine but maybe I did not understand what you mean?

Reply:
I mean that if something regresses and starts giving a shorter string, this test will warm but still pass. It should definitely fail if the strings are different lengths...

Reply:
Ok, the test will fail just after in the for loop when it will compare portions of the expected and the actual. Even if one of the actual or expected string are empty.

Reply:
Right, but you're only looping through up to strSize, which is defined as actualStr.size(). So if the expectedStr.size() is greater than the actualStr.size(), you'll never compare those missing bits of actualStr to what they should be. 

I think line 319 should be `const int strSize = std::max( actualStr.size(), expectedStr.size() );`

Reply:
Ok @nyalldawson! I have fixed that and made little changes in output.